src/main/java/in/projecteka/consentmanager/link/discovery/Discovery.java

```diff
@@ -63,6 +75,63 @@
                                         requestId)));
     }
 
+    public Mono<DiscoveryResponse> patientInHIP(String userName,
+                                              List<PatientIdentifier> unverifiedIdentifiers,
+                                              String providerId,
+                                              UUID transactionId,
+                                              UUID requestId) {
+        return Mono.just(requestId)
+                .filterWhen(this::validateRequest)
+                .switchIfEmpty(Mono.error(ClientError.requestAlreadyExists()))
+                .flatMap(val ->
+                        userWith(userName)
+                                .zipWith(gatewaySystemUrl())
+                                .flatMap(userAndGateway ->
+                                        discoveryServiceClient.requestPatientFor(
+                                                requestFor(userAndGateway.getT1(), transactionId, unverifiedIdentifiers),
+                                                userAndGateway.getT2(),
+                                                providerId))
+                                .zipWith(Mono.delay(Duration.ofSeconds(getExpectedFlowResponseDuration())))
```

Comment:
Based on the doc of zipwith, it seems like it might be cancelled when other Mono returns empty. Please check if usage of zipwith is still valid.

Reply:
yep. returning Mono<Boolean> in the previous Mono method.